### PR TITLE
Don't raise a warning when `nn.Module`s are not saved under hparams

### LINKED ISSUE #12664: `save_hyperparameters` gives a warning incorrectly
## 🐛 Bug

Save hyperparameters gives a warning about nn.Modules incorrectly when only saving non-module attributes via the `*args`.

### To Reproduce

```python
from torch import nn
from pytorch_lightning import LightningModule


class MyLightningModule(LightningModule):

    def __init__(self, backbone, head, learning_rate):
        super().__init__()

        # Only save the `learning_rate`
        self.save_hyperparameters("learning_rate")

        self.backbone = backbone
        self.head = head
        self.learning_rate = learning_rate


# This line gives two warnings
_ = MyLightningModule(nn.Linear(10, 10), nn.Linear(10, 10), 0.1)
```

### Expected behavior

Shouldn't raise an error as I beleive this example is only saving the learning rate.

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes #12664

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]
